langchain/src/embeddings/bedrock.ts

```diff
@@ -93,13 +115,23 @@ export class BedrockEmbeddings
   }
 
   /**
-   * Method that takes an array of documents as input and returns a promise
-   * that resolves to a 2D array of embeddings for each document. It calls
-   * the _embedText method for each document in the array.
-   * @param documents Array of documents for which to generate embeddings.
+   * Method to generate embeddings for an array of texts. Splits the
+   * texts into batches and makes requests to Bedrock to generate
+   * embeddings.
+   * @param documents Array of texts for which to generate embeddings.
    * @returns Promise that resolves to a 2D array of embeddings for each input document.
    */
-  embedDocuments(documents: string[]): Promise<number[][]> {
-    return Promise.all(documents.map((document) => this._embedText(document)));
+  async embedDocuments(documents: string[]): Promise<number[][]> {
+    const batches = chunkArray(documents, this.batchSize);
+    const embeddings: number[][] = [];
+
+    for (const batch of batches) {
```

Comment:
You don't actually have to batch here - just `Promise.all` everything. The `caller` will take care of concurrency.

Reply:
https://github.com/langchain-ai/langchainjs/blob/main/langchain/src/embeddings/openai.ts#L159